src/sciwrappers.c

```diff
@@ -145,12 +145,31 @@ void sci_set_mark_long_lines(ScintillaObject *sci, gint type, gint column, const
 }
 
 
+/* compute margin width based on ratio of line height */
+static gint margin_width_from_line_height(ScintillaObject *sci, gdouble ratio, gint threshold)
+{
+	gint line_height = SSM(sci, SCI_TEXTHEIGHT, 0, 0);
+	gint width;
+
+	width = line_height * ratio;
+	/* round down to an even size */
+	width = width - (width % 2);
+	/* if under threshold, just use the line height */
+	if (width < threshold)
+		width = MIN(threshold, line_height);
+
+	return width;
+}
+
+
 /* symbol margin visibility */
 void sci_set_symbol_margin(ScintillaObject *sci, gboolean set)
 {
 	if (set)
 	{
-		SSM(sci, SCI_SETMARGINWIDTHN, 1, 16);
+		gint width = margin_width_from_line_height(sci, 0.88, 16);
```

Comment:
`0.88` come from `16/18`, where 18 was the line height of the pretty regular font size on my system, and 16 the previous width.

Reply:
I believe modern compilers will optimize constant local variables out completely.

Reply:
@b4n, well, 18 might be right on your monitor, but its probably 36 on @codebrainz 4k monitor.

Reply:
@codebrainz made const.

@elextr ? that's the point, we currently hardcode 16px, which is very small if you use a big font/zoom level.  For HiDPI the surface should be scaled in the best case (well, best case Scintilla would handle some of it itself I guess, but well), but otherwise this change just makes it better there too.